pipeline-yaml_5b5744d40428631d7a8940a9.md

```diff
@@ -0,0 +1,548 @@
+### agent
+
+The `agent` property requires three additional properties that identify the
+environment, that is the hardware of the Virtual machine and the Operating
+System of the Virtual Machine, in which the jobs of the pipeline will get
+executed.
+
+Example:
+
+    agent:
+      machine:
+        type: e1-standard-2
+        os_image: ubuntu1804
+
+
+#### machine
+
+The `machine` property, which can only be found under `agent`, holds the `type`
+and `os_image` properties.
+
+Example:
+
+    machine:
+      type: e1-standard-2
+      os_image: ubuntu1804
+
+
+#### type
+
+The `type` property is intended for selecting the machine (hardware) you would
+like to use for your builds.
+
+List of valid values: `e1-standard-2`
+
+Example: `type = e1-standard-2`
+
+
+#### os_image
+
+The `os_image` property specifies the operating system and the version of the
+operating system that will be used.
+
+
+Valid values: `ubuntu1804`
+
+
+Example: `os_image = ubuntu1804`
+
+#### Constants
+
+##### The e1-standard-2 constant
+
+The `e1-standard-2` value used in the preamble represents the hardware of
+the Virtual Machine used for each job in the current pipeline and is a global
+variable.
+
+
+##### The ubuntu1804 constant
+
+The `ubuntu1804` value used in the preamble represents the Operating System of
+the Virtual Machine used for each job in the current pipeline and is a
+global variable.
+
+
+
+### Preamble example
+
+    version: "v1.0"
+    name: "The name of the Semaphore 2.0 project"
+    agent:
+      machine:
+        type: e1-standard-2
+        os_image: ubuntu1804
+    blocks:
+
+
+## Blocks
+
+The `blocks` property is an array of `task` items that hold the elements of
+a `.semaphore/semaphore.yml` file.
+
+The values of the `blocks` property are the tasks of a pipeline as
+defined by the `task` property.
+
+If you accidentally name two or more `blocks` items with the same name,
+you will get an error message similar to the following:
+
+> semaphore.yml ERROR:
+
+> Error: "There are at least two blocks with same name: Build Go project"
+
+
+### name
+
+The `name` property is a Unicode string that assigns a name to the Semaphore
+2.0 project and is optional.
+
+
+### task
+
+All the items of a `blocks` list of of the `task` type.
+
+You will learn about the properties of a `task` item in a while.
+
+
+### Example of Blocks
+
+    blocks:
+     - name: Inspect Linux environment
+       task:
+          jobs:
+            - name: Execute hw.go
+              commands:
+                - echo $SEMAPHORE_PIPELINE_ID
+                - echo $HOME
+                - echo $SEMAPHORE_GIT_DIR
+                - pwd
+                - ls -al
+                - cat hw.go
+                - echo $VAR1
+                - echo $PI
+          prologue:
+              commands:
+                - checkout
+          env_vars:
+               - name: VAR1
+                 value: Environment Variable 1
+               - name: PI
+                 value: "3.14159"
```

Comment:
For this example is overly verbose for the purpose it's serving. 

- There is not need to have 8 commands. One or two are fine. 
- We don't need `prologue` and `env_vars` in example purely focused on task.

Keep example in runnable form and to a bare minimum to present the `task` property.

Reply:
Made it much smaller without `prologue` and `env_vars`.